contrib/src/main/java/com/datatorrent/contrib/parser/FixedWidthValidator.java

```diff
@@ -0,0 +1,473 @@
+    try {
+
+      if (value != null && !value.isEmpty()) {
+        Float result = Float.parseFloat(value);
+        toEmit.put(fieldName,result);
+        if (typeInfo != null && pojoObject != null) {
+          typeInfo.setter.set(pojoObject, result);
+        }
+      }
+      else
+        toEmit.put(fieldName, value);
+    } catch (NumberFormatException e) {
+      throw new RuntimeException("Error parsing" + value + " to Float type", e);
+    } catch (Exception e) {
+      throw new RuntimeException("Error setting " + value + " in the given class" + typeInfo.toString(), e);
+
+    }
+  }
+
+  /**
+   * Function to validate Long type data and set it in the Object
+   * using accessors in TypeInfo
+   * @param value
+   * @param fieldName
+   * @param typeInfo
+   * @param pojoObject
+   * @param toEmit
+   */
+  public void parseandsetLongValue(String value, String fieldName, FixedWidthParser.TypeInfo typeInfo, Object pojoObject, HashMap toEmit)
+  {
+    try {
+      if (value != null && !value.isEmpty()) {
+        Long result = Long.parseLong(value);
+        toEmit.put(fieldName,result);
+        if (typeInfo != null && pojoObject != null) {
+          typeInfo.setter.set(pojoObject, result);
+        }
+      }
+      else
+        toEmit.put(fieldName, value);
+    } catch (NumberFormatException e) {
+      throw new RuntimeException("Error parsing" + value + " to Long type", e);
+    } catch (Exception e) {
+      throw new RuntimeException("Error setting " + value + " in the given class" + typeInfo.toString(), e);
+
+    }
+
+  }
+
+  /**
+   *  Function to validate Short type data and set it in the Object
+   * using accessors in TypeInfo
+   * @param value
+   * @param fieldName
+   * @param typeInfo
+   * @param pojoObject
+   * @param toEmit
+   */
+  public void parseandsetShortValue(String value, String fieldName, FixedWidthParser.TypeInfo typeInfo, Object pojoObject, HashMap toEmit)
+  {
+    try {
+      if (value != null && !value.isEmpty()) {
+        Short result = Short.parseShort(value);
+        toEmit.put(fieldName,result);
+        if (typeInfo != null && pojoObject != null) {
+          typeInfo.setter.set(pojoObject, result);
+        }
+      }
+      else
+        toEmit.put(fieldName, value);
+    } catch (NumberFormatException e) {
+      throw new RuntimeException("Error parsing" + value + " to Short type", e);
+    } catch (Exception e) {
+      throw new RuntimeException("Error setting " + value + " in the given class" + typeInfo.toString(), e);
+
+    }
+
+  }
+
+  /**
+   * Function to validate Boolean type data and set it in the Object
+   * using accessors in TypeInfo
+   * @param value
+   * @param fieldName
+   * @param trueValue
+   * @param falseValue
+   * @param typeInfo
+   * @param pojoObject
+   * @param toEmit
+   */
+  public void parseandsetBooleanValue(String value, String fieldName, String trueValue, String falseValue, FixedWidthParser.TypeInfo typeInfo, Object pojoObject, HashMap toEmit)
```

Comment:
Can you merge all these different functions into one and use a switch case or an if/else to handle all these cases?